### ISSUE TITLE
[Bug]: Course name for courses like CS 4V95

### ISSUE THREAD

**Issue by AbhiramTadepalli** [UTDNebula/utd-trends#183]
### What happened?

CS 4V95 is a "special topics" course: "Undergraduate Topics in Computer Science Software Engineering". Right now, it shows _"Undergraduate Topics in Computer Science - Network Traffic Generation"_ as the course's name; this is one instance of the class, but doesn't represent the entire class.

### What should happen
Any course that has a variable (special topic) should have a general course name, not the specific instance of the course that comes after the dash _in this case_

- [x] Identify all the "special topics" courses like (variable number of credit hours, like CS 2V95)
- [x] Find a way to remove the specific info from the course name
- [ ] Verify it works for all courses like this

### Version

Beta 0.1.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by egsch** [comment]
Courses with variable credit hours that are showing course section titles instead of general class titles:
```SE 4V95 - Undergraduate Topics in Software Engineering - Personality Capture and Emulat
VS 4V98 - Undergraduate Research in Computer Science - Undergraduate Research in Cs
EE 4V98 Undergraduate Research in Electrical Engineering - Cochlear Implant Research
CE 4V95 - Undergraduate Research in Electrical Engineering - Cochlear Implant Research
CS 4V95 - Undergraduate Research in Electrical Engineering - Cochlear Implant Research
ED 4V95 - Special Topics in Education - Exceptional Populations
EE 4V95 - Undergraduate Topics in Computer Engineering - Fundamentals of Power Systems
BA 4V00 - Special Topics - 4350 Global Mkt Entry Strategy
FIN 4V90 - Individual Study in Finance - Profitability Tech Aviat Mkt
NSC 4V90 - Special Topics in Neuroscience - Tauopathies
IMS 3V92 - Regional Management Area Studies Western Europe - Rome, Italy
IMS 3V94 - Regional Management Area Studies Africa - Rwanda
IMS 3V93 - Regional Management Area Studies Asia - Tokyo, Japan
BIOL 3V00 - Topics in Biological Sciences - Biomedical Systems & Modeling
HONS 3V00 - Special Topics - CV Honors:primer Endocrinology
Mech 3V95 - Topics in Mechanical Engineering - Materials Sci Sustainable Engy
LANG 2V42 - Topics in Language - Spanish Language Internship
UNIV 2V96 - Special Topics I - First Gen Student Success
```

These are all separated by " - ", but we also have some other courses with " - " in the course title like `BPS 4395 - Capstone Senior Project - Business` so simply splitting by " - " won't work. 

Is there a reason we use the course title from the course section instead of extracting it with the description from the scraped Coursebook data?

**Comment by egsch** [comment]
Update after looking into this more, sorry this is a lot for a small bug:
- Most calls to the course API for special topics courses return at least one course with a general title/# of credit hours. However, these usually aren't from the most recent catalog year, so when we sort the response array we get one with a specific section title and (sometimes) a specific # of credit hours.
  - When we're parsing the description for a course in the second category, having a specific credit hour # instead of a range can break the description similar to the special character issue.

Obvious possible solutions:
- Filter for the older general course catalog listings and hope nothing's changed. May not fix credit hour issue.
- Parse the course titles and remove the part after the ' - ' only if hours are variable. Still leaves credit hour issue.
- Tell the API team about the issue and get them to standardize future course names/credit hour listings. Might be a good idea, but won't solve the issue for the release.
- Replace our current parsing algorithm with something like this and return title/credit hours to use in component (this would also fix #173 but might cause other issues):
```
const [descriptionBeforeCreditHours, descriptionAfterCreditHours] = course.description.split(/\([\d\-]{0,3} semester credit hours\)/);
let formattedDescription = descriptionAfterCreditHours || course.description;
const courseTitle = course.course_number[1] =='V' ? descriptionBeforeCreditHours.split(course.course_number + " - ")[1] : course.title;
const creditHours = course.course_number[1] =='V' ? course.description.split(" semester credit hours\)")[0].split("(")[1] : course.credit_hours;
```

**Comment by AbhiramTadepalli** [comment]
Actually I think the last one (parsing the description for the title) should work best. We can default to the title API gives us if we are unable to identify the title from the descriptions.

Go ahead and implement this and we can test
